roles/spark/client/tasks/install.yml

```
@@ -28,6 +28,16 @@
   when: spark_hbase_connector_enable and 'hbase_client' in groups
   diff: false
 
+- name: Upload iceberg-spark-runtime jar
+  ansible.builtin.copy:
+    src: "{{ binaries_local_dir }}/{{ iceberg_spark_runtime_dist_file }}"
+    dest: "{{ spark_root_dir }}/{{ spark_release }}/jars"
+    owner: root
+    group: root
+    mode: "755"
+  diff: false
+  when: spark_iceberg_enable
```

spark2 install failed because `spark_iceberg_enable` is not defined for spark2. 
```suggestion
  when: spark_iceberg_enable | default(false)
```

---

It is actually going to be the other way around, by default it is going to be true. I still have to change this in `tdp_vars_default`.